### PR TITLE
add rhui and devtools rpms for fips go-toolset

### PR DESCRIPTION
**Pull request by bryanro92**
### Which issue this PR addresses:

Fixes https://msazure.visualstudio.com/AzureRedHatOpenShift/_workitems/edit/10354848/

### What this PR does / why we need it:

downloads and sets cloud-user to use the fips compliant go-toolset 
### Test plan for issue:

tested by spinning up azure vm, and running scripts to confirm results are as expected. would like to deploy to CI vm's as well to confirm changes.

### Is there any documentation that needs to be updated for this PR?
no

### CODE REVIEW

**Comment by bryanro92** [comment]
Adding hold label until I can deploy this PR to INT environment to validate changes.

**Review by petrkotas — DISMISSED** [review]
Hi @bryanro92 I like the simplicity of this PR, really clean job. I have found one nit for you to consider. Otherwise it is lgtm, once it is tested.

File: pkg/deploy/generator/resources_dev.go [hunk 1]
```diff
@@ -336,8 +336,8 @@ sudo -u cloud-user ./config.sh --unattended --url https://dev.azure.com/msazure
 popd
 
 # merge in /opt/rh/rh-*/enable
-cat >/home/cloud-user/agent/.path <<'EOF'
-/opt/rh/rh-python36/root/usr/bin:/opt/rh/rh-git29/root/usr/bin:/usr/local/bin:/usr/bin:/usr/local/sbin:/usr/sbin:/home/cloud-user/.local/bin:/home/cloud-user/bin
+cat <<EOF > /home/cloud-user/agent/.path
```
Comment (mjudeikis) [inline pkg/deploy/generator/resources_dev.go:339]: we use `cat >fil <<EOF` in most places. Should we do it here too? Or is there good reason for other pattern, if so we should change it everywhere.

Comment (bryanro92) [reply pkg/deploy/generator/resources_dev.go]: I have changed back to that pattern now knowing it is preferred 😀

File: pkg/deploy/generator/resources_dev.go [hunk 2]
```diff
@@ -318,11 +318,11 @@ enabled=yes
...
-yum --enablerepo=rhui-rhel-7-server-rhui-optional-rpms -y install azure-cli docker jq gcc rh-git29 rh-python36 tmpwatch
-
-GO_VERSION=1.14.9
-curl https://dl.google.com/go/go${GO_VERSION}.linux-amd64.tar.gz | tar -C /usr/local -xz
-ln -s /usr/local/go/bin/* /usr/local/bin
+GO_VERSION=1.14
+yum-config-manager --enable rhel-7-server-devtools-rpms 
+yum-config-manager --enable rhui-rhel-7-server-optional-rpms 
+yum-config-manager --enable rhui-rhel7-server-rpms
+yum -y install azure-cli docker jq gcc rh-git29 rh-python36 tmpwatch go-toolset-$GO_VERSION
```
[the remark this answers is not in the document]
Comment (bryanro92) [reply pkg/deploy/generator/resources_dev.go:325]: Operating System has OpenSSL for base crypto libs, and go-toolset comes with its own FIPS validated ssl ones that will be used for any `go` commands

File: pkg/deploy/generator/resources_dev.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (bryanro92) [reply pkg/deploy/generator/resources_dev.go]: I have updated as requested @petrkotas 😀

**Comment by bryanro92** [comment]
still hitting a blocker with testing this. It turns out the devtools-rpms is not available in RHUI rpms, meaning it is locked behind a subscription which would require a BYOS in current model to fix this. Looking into RHEL8 and other solutions that my not require FIPS on the host if entire CI build process is moved into containers.

**Comment by bryanro92** [comment]
closing in favor of #1774; thanks @bennerv

**Pull request closed without merging**